__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+    }
+
+    // Validating interfaces with all resources unlocked
+    const waitFor2 = timelock2 - Date.now().valueOf() + 1000;
+    loggers.test.log(`Will wait for ${waitFor2}ms for timelock2 to expire`);
+    await delay(waitFor2);
+
+    // Forcing balance updates
+    await hWallet.preProcessWalletData();
+
+    // Validating getBalance ( moment 2 )
+    htrBalance = await hWallet.getBalance(HATHOR_TOKEN_CONFIG.uid);
+    expect(htrBalance[0].balance.locked).toBe(0);
+    expect(htrBalance[0].balance.unlocked).toBe(10);
+
+    // Confirm that now the balance is available
+    const sendTx = await hWallet.sendTransaction(hWallet.getAddressAtIndex(4), 8);
+    expect(sendTx).toHaveProperty('hash');
+  });
+});
+
+describe('createNewToken', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should create a new token', async () => {
+    // Creating the wallet with the funds
+    const hWallet = await generateWalletHelper();
+    const addr0 = hWallet.getAddressAtIndex(0);
+    await GenesisWalletHelper.injectFunds(addr0, 10);
+
+    // Creating the new token
+    const newTokenResponse = await hWallet.createNewToken(
+      'TokenName',
+      'TKN',
+      100,
+    );
+
+    // Validating the creation tx
+    expect(newTokenResponse).toHaveProperty('hash');
+    expect(newTokenResponse).toHaveProperty('name', 'TokenName');
+    expect(newTokenResponse).toHaveProperty('symbol', 'TKN');
+    expect(newTokenResponse).toHaveProperty('version', 2);
+    const tokenUid = newTokenResponse.hash;
+
+    // Validating wallet balance is updated with this new token
+    await waitForTxReceived(hWallet, tokenUid);
+    const tknBalance = await hWallet.getBalance(tokenUid);
+    expect(tknBalance[0].balance.unlocked).toBe(100);
+  });
+
+  it('should create a new token on the correct addresses', async () => {
+    // Creating the wallet with the funds
+    const hWallet = await generateWalletHelper();
+    const addr0 = hWallet.getAddressAtIndex(0);
+    await GenesisWalletHelper.injectFunds(addr0, 10);
+
+    // Creating the new token
+    const destinationAddress = hWallet.getAddressAtIndex(4);
+    const changeAddress = hWallet.getAddressAtIndex(8);
+    const { hash: tokenUid } = await hWallet.createNewToken(
+      'NewToken Name',
+      'NTKN',
+      100,
+      {
+        address: destinationAddress,
+        changeAddress
+      }
+    );
+    await waitForTxReceived(hWallet, tokenUid);
+
+    // Validating the tokens are on the correct addresses
+    const { utxos: utxosTokens } = hWallet.getUtxos({ token: tokenUid });
+    expect(utxosTokens.find(u => u.address === destinationAddress))
+      .toHaveProperty('amount', 100);
+
+    const { utxos: utxosHtr } = hWallet.getUtxos();
+    expect(utxosHtr.find(u => u.address === changeAddress))
+      .toHaveProperty('amount', 9);
+  });
+
+  // TODO: Test creating token without mint/melt authority
```

Comment:
Yes, this is important.

Reply:
An issue will be opened to implement this feature on the `createNewToken` method.

Reply:
Working on it!

Reply:
This test was built on [353b5e10](https://github.com/HathorNetwork/hathor-explorer-service/pull/174/commits/353b5e10cf2ae82e2fe25c7f507327000a0fb6c4). I also had to rewrite some of the JSDocs to better document this feature.